# **ML - Task2 - Hypothesis testing**
* **Q1.Suppose a child psychologist claims that the average time working mothers spend talking to their children is at least 11 minutes per day. You conduct a random sample of 1000 working mothers and find they spend an average of 11.5 minutes per day talking with their children. Assume prior research suggests the population standard deviation is 2.3 minutes.Conduct a test with a level of significance of alpha = 0.05**

# **Null Hypothesis**
**Ho = μ ≤ 11 minutes**

# **Alternative Hypothesis**
**Ha = μ > 11 minutes**

**x̄ = 11.5 minutes**

**n = 1000**

**σ = 2.3 minutes**

**α = 0.05**

In [ ]:
import math





In [ ]:
alpha = 0.05

In [ ]:
std = 2.3

In [ ]:
n =1000

In [ ]:
n_mean = 11.5

In [ ]:
populataion_mean =11

In [ ]:
#Hypothesis testing

z_test = (n_mean - populataion_mean) / (std/math.sqrt(n))

In [ ]:
z_test

6.874516652539955

In [ ]:
import scipy.stats as st


alpha = 0.05


z_critical = st.norm.ppf(1 - alpha)

print("Critical value:", z_critical)





Critical value: 1.6448536269514722


#**conclusion**
*  **The z test value is greater than the critical value so we reject the null hypothesis and conclude that the working mother spend more than 11 minutes talking with their children per day**

* **Q2. A coffee shop claims that their average wait time for customers is less than 5 minutes. To test this claim, a sample of 40 customers is taken, and their wait times are recorded. The sample mean wait time is found to be 4.6 minutes with a standard deviation of 0.8 minutes. Perform a hypothesis test at a significance level of 0.05 and determine whether there is enough evidence to support the coffee shop's claim**

# **Null Hypothesis**

Ho = μ ≥ 5 minutes

# **Alternative Hypothesis**

Ha = μ < 5 minutes

Sample mean(x̄) = 4.6 minutes

Sample size(n) = 40

Sample Standard deviation(σ) = 0.8 minutes

Population mean(μ) = 5 minutes

Significance level(α) = 0.05(5%)

Degree of freedom(n-1) = 39

In [ ]:
population_tmean = 5

In [ ]:
sample_tmean =4.6

In [ ]:
sample_size = 40

In [ ]:
std_t = 0.8

In [ ]:
alpha = 0.05


* **The t-test is used in this case because we are comparing the mean of a sample (the average wait time of customers in the coffee shop) to a known or hypothesized population mean (5 minutes) when the population standard deviation is unknown.**

In [ ]:
t_test =  (sample_tmean - population_tmean) / (0.8/math.sqrt(sample_size))

In [ ]:
t_test

-3.162277660168382

In [ ]:
from scipy.stats import t

# Degrees of freedom
df = 40 - 1  # Sample size minus 1


alpha = 0.05

# Find critical value
t_critical = t.ppf(1 - alpha, df)

print("Critical value:", t_critical)

Critical value: 1.6848751194973992


**Conclusion:** The calculated t-value (−3.1623) is less than the critical value (
1.68491). Therefore, we reject the null hypothesis. There is enough evidence to support the coffee shop's claim that their average wait time for customers is less than 5 minutes.






